File: packages/aws-cdk/lib/init-templates/v1/app/java/src/test/java/com/myorg/%name.PascalCased%Test.template.java
package com.myorg;

import software.amazon.awscdk.core.App;
import software.amazon.awscdk.assertions.Template;
import software.amazon.awscdk.assertions.Match;
import java.io.IOException;

import java.util.*;

import org.junit.jupiter.api.Test;

public class %name.PascalCased%Test {

    @Test
    public void testStack() throws IOException {
        App app = new App();
        %name.PascalCased%Stack stack = new %name.PascalCased%Stack(app, "test");

        Template template = Template.fromStack(stack);

        // Update once resources have been added to the stack
        Map<String, Object> expected = Map.of();
        template.templateMatches(expected);
    }
}

File: packages/aws-cdk/lib/init-templates/v1/app/kotlin/src/test/kotlin/com/myorg/%name.PascalCased%Test.template.kt
package com.myorg

import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import software.amazon.awscdk.core.App

class %name.PascalCased%Test {
    private val mapper = ObjectMapper().configure(SerializationFeature.INDENT_OUTPUT, true)

    @Test
    fun testStack() {
        val app = App()
        val stack = %name.PascalCased%Stack(app, "test")

        // synthesize the stack to a CloudFormation template
        val actual = mapper.valueToTree<JsonNode>(app.synth().getStackArtifact(stack.artifactId).template)

        // Update once resources have been added to the stack
        Assertions.assertThat(actual["Resources"]).isNull()
    }
}
